cranelift/codegen/src/machinst/isle.rs

```diff
@@ -122,11 +122,28 @@ macro_rules! isle_lower_prelude_methods {
 
         #[inline]
         fn put_in_reg(&mut self, val: Value) -> Reg {
-            self.lower_ctx.put_value_in_regs(val).only_reg().unwrap()
+            self.put_in_regs(val).only_reg().unwrap()
         }
 
         #[inline]
         fn put_in_regs(&mut self, val: Value) -> ValueRegs {
+            // If the value is a constant, then (re)materialize it at each
+            // use. This lowers register pressure.
```

Comment:
Does this mean that we now have this rematerialization behavior unconditionally, i.e. we lost the not-egraph-mode condition? (I see that in your commit message but it's slightly unclear if it's replaced by a check anywhere else that I might be missing.) The reason for this was that the egraph-based mid-end does its own rematerialization handling, and will replicate iconsts/fconsts into different blocks as appropriate, so we want to avoid this redundant behavior. (Ultimately once egraph-based optimization is on by default we can remove this remat code here entirely.)

Reply:
Ah, I see, I may have misinterpreted the logic.  I've added the egraphs check back here.